File: May2021/EvaluateReversePolishNotation.java
class Solution {
    // TC  : O(n)

    public int evalRPN(String[] tokens) {
        Stack<String> st = new Stack<>();

        for(String el: tokens){
            if(isOperator(el)){

                int el2 = Integer.parseInt(st.pop());
                int el1 = Integer.parseInt(st.pop());
                int ans = 0;

                if(el.equals("*")){
                    ans = el1 *el2;
                } else if(el.equals("/")){
                    ans = el1/el2;
                }else if(el.equals("+")){
                    ans = el1 +el2;
                }else if(el.equals("-")){
                    ans = el1 -el2;
                }
                st.push(ans+"");
            } else {
                // operand
                st.push(el);
            }
        }

        return Integer.parseInt(st.peek());
    }

    private boolean isOperator(String el){
        if(el.equals("*") || el.equals("+") || el.equals("-") || el.equals("/")){
            return true;
        } else {
            return false;
        }
    }
}

File: December2022/C++/evaluate-reverse-polish-notation.cpp
#define ll long long int
class Solution {
public:
   int evalRPN(vector<string>& tokens) {
    
       stack<ll>st;
       
       ll len = tokens.size();
       
       for(auto num:tokens){
           
           if(num.size()>1 || isdigit(num[0])){
               
               st.push(stoll(num));
           }
           else{
               
               auto x2 = st.top(); st.pop();
               auto x1 = st.top(); st.pop();
               
               switch(num[0]) {
                case '+': x1+=x2; break;
                case '-': x1-=x2; break;
                case '*': x1*=x2; break;
                case '/': x1/=x2; break;
            }
            st.push(x1);
               
           }
       }
       
       return st.top();
}
};